### PR TITLE
tests: protoassert.SliceContains

### PR DESCRIPTION
**Pull request by janisz**
- #11531
- #11533
- #11503
- #11507
- #11575
- #11571 :point_left: 
- #11572 
- https://github.com/stackrox/stackrox/pull/11582
- https://github.com/stackrox/stackrox/pull/11583
- https://github.com/stackrox/stackrox/pull/11585
- https://github.com/stackrox/stackrox/pull/11588  
- https://github.com/stackrox/stackrox/pull/11618 
- https://github.com/stackrox/stackrox/pull/11619

### CODE REVIEW

**Review by rhybrillou — COMMENTED** [review]
Could you please rebase this PR ?

File: pkg/protoassert/assert.go [hunk 1]
```diff
@@ -40,6 +40,17 @@ func SlicesEqual[T proto.Message](t testing.TB, expected, actual []T, msgAndArgs
...
 }
 
+func SliceContains[T proto.Message](t testing.TB, slice []T, element T, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	for _, e := range slice {
+		if proto.Equal(e, element) {
+			return true
+		}
+	}
+	t.Fatalf("Slice does not contain element: %s %v", element.String(), msgAndArgs)
```
Comment (rhybrillou) [inline pkg/protoassert/assert.go:50]: `t.Fatalf` will interrupt the test execution (more or less the `require` variant of an assertion).
Do we want that or should `t.Errorf` (non-interrupting variant) be called instead ?

File: central/alert/service/service_test.go
Comment (rhybrillou) [inline central/alert/service/service_test.go:1]: It looks like the changes in this file are not related to the change at hand.

**Review by rhybrillou — COMMENTED** [review]
I did review the commit with the same name as the PR. There is one comment on the assert function implementation.
For the rest, I think the PR has leftovers from some rebases. Could you make sure the stack of PRs is properly rebased and the diff to review matches the description and expectations (I suppose the diff here should rather be 5 or 7 files than the current 200).

File: pkg/protoassert/assert.go (hunk 1 above)
Comment (janisz) [reply pkg/protoassert/assert.go]: Good catch, I fixed that with `assert.Fail`

File: pkg/protoassert/assert.go [hunk 2]
```diff
@@ -40,6 +40,16 @@ func SlicesEqual[T proto.Message](t testing.TB, expected, actual []T, msgAndArgs
...
 }
 
+func SliceContains[T proto.Message](t testing.TB, slice []T, element T, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	for _, e := range slice {
+		if proto.Equal(e, element) {
+			return true
+		}
+	}
+	return assert.Failf(t, "Slice does not contain element: %s %v", element.String(), msgAndArgs)
```
Comment (rhybrillou) [inline pkg/protoassert/assert.go:50]: The signature of `Failf` is:
`func Failf(t TestingT, failureMessage string, msg string, args ...interface{}) bool`

I think you need a generic message as first argument, and not a format string.

Tested locally, the following code:
```
package test

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestFailf(t *testing.T) {
	assert.Failf(t, "message template: %s", "hello")
}
```
gives this result:
```
=== RUN   TestFailf
    tst_test.go:10: 
        	Error Trace:	/path/to/tst_test.go:10
        	Error:      	message template: %s
        	Test:       	TestFailf
        	Messages:   	hello
--- FAIL: TestFailf (0.00s)
```

Comment (janisz) [reply pkg/protoassert/assert.go]: You're right, fixed

```go
package protocompat

import (
	"testing"

	"github.com/stackrox/rox/pkg/protoassert"
)

func TestEmpty(t *testing.T) {
	refEmpty := &Empty{}
	protoassert.SliceContains(t, []*Empty{}, refEmpty, "some message", "and", "args")
}

```

```
        	Error:      	Slice does not contain element
        	Test:       	TestEmpty
                Messages:   	"" [some message and args]
```

**Review by rhybrillou — APPROVED** [review]

**Review by mtodor — APPROVED** [review]
